testsuite/tests/kuadrant/observability/test_tracing.py

```diff
@@ -0,0 +1,23 @@
+"""Tests for Kuadrant tracing"""
+
+import os
+from time import sleep
+
+
+def test_tracing(client, auth, tracing):
+    """
+    Tests that traces are collected from Authorino and Limitador
+
+    Trace IDs do not propagate to wasm modules in Istio/Envoy, affecting trace continuity in Limitador:
+    https://github.com/envoyproxy/envoy/issues/22028
+    That is the reason why we set parent trace ID in request.
+    """
+    trace_id = os.urandom(16).hex()
+    client.get("/get", auth=auth, headers={"Traceparent": f"00-{trace_id}-{os.urandom(8).hex()}-01"})
+    sleep(5)  # Waits for tracing backend to collect all traces
```

Comment:
Why 5 and is there not a better waiting mechanism? Maybe retry?

Reply:
It takes some time for the collector to get all the traces. What happens here, is that the tracing backend correctly returns the trace by `tracreId` but it takes some time for this trace to contain all spans.